src/libs/auth.ts

```diff
@@ -97,49 +100,68 @@ const revokeTokens = async () => {
 const getSigninArgs = (includeBillingScope) => {
   return Utils.cond(
     [includeBillingScope === false, () => ({})],
-    // For Google just append the scope to the signin args.
-    [isGoogleAuthority(), () => ({ scope: 'openid email profile https://www.googleapis.com/auth/cloud-billing' })],
-    // For B2C switch to a dedicated policy endpoint configured for the GCP cloud-billing scope.
+    // For B2C use a dedicated policy endpoint configured for the GCP cloud-billing scope.
     () => ({
       extraQueryParams: { access_type: 'offline', p: getConfig().b2cBillingPolicy },
       extraTokenParams: { p: getConfig().b2cBillingPolicy },
     })
   );
 };
 
-export const signIn = async (includeBillingScope = false) => {
+export const signIn = async (includeBillingScope = false): Promise<User> => {
   const args = getSigninArgs(includeBillingScope);
-  const user = await getAuthInstance().signinPopup(args);
-
-  // For B2C record in the auth store whether we requested the GCP cloud-billing scope since there
-  // is no way to determine it after the fact.
-  // For Google we don't need to do this since we can inspect the scope directly in the user object.
-  if (!isGoogleAuthority()) {
-    authStore.update((state) => ({ ...state, hasGcpBillingScopeThroughB2C: includeBillingScope }));
-  }
-  Ajax().Metrics.captureEvent(Events.userLogin, { authProvider: user.profile.idp });
+  const user: User = await getAuthInstance().signinPopup(args);
+
+  const sessionId = uuid();
+  const sessionStartTime: number = Date.now();
+  const userJWT: any = user.id_token;
+  const decodedJWT: JwtPayload = jwtDecode<JwtPayload>(userJWT);
+  const authTokenCreatedAt: any = (decodedJWT as any).auth_time; // time in seconds when authorization token was created
+  const authTokenExpiresAt: any = user.expires_at; // time in seconds when authorization token expires
+  const jwtTokenExpiresAt = (decodedJWT as any).exp; // time in seconds when jwt expires (should not be read from)
```

Comment:
```suggestion
  const authTokenCreatedAt: number = (decodedJWT as any).auth_time; // time in seconds when authorization token was created
  const authTokenExpiresAt: number = user.expires_at; // time in seconds when authorization token expires
  const jwtTokenExpiresAt: number = (decodedJWT as any).exp; // time in seconds when jwt expires (should not be read from)
```

It sounds like we know the expected types here, so let's us the specific type instead of `any`.

Reply:
Also, could we be more consistent with naming? Two variables are prefixed with `authToken`, one with `jwtToken`. This suggests that the variables come from different sources, but actually `authTokenCreatedAt` and `jwtTokenExpiresAt` are both taken from the same token.

And nit: JWT stands for "JSON Web Token", so `jwtToken` reads as "JSON Web Token Token".

Reply:
What is the difference between `authTokenExpiresAt` and `jwtTokenExpiresAt`?

Reply:
My understanding is that the `authTokenExpires` is whatever the oidc provider sends back as the expiration time of the authToken. The `jwtTokenExpiresAt` is when the jwt itself expires and should not be read from anymore. I think right now they both are 1 hour as far as I've seen, but that is not a guarantee, so I thought it might be helpful to log in case there is any discrepancies.